rosbag2_tests/test/rosbag2_tests/test_rosbag2_record_end_to_end.cpp

```diff
@@ -70,6 +70,67 @@ TEST_F(RecordFixture, record_end_to_end_test) {
   EXPECT_THAT(wrong_topic_messages, IsEmpty());
 }
 
+TEST_F(RecordFixture, record_end_to_end_with_splitting_splits_bagfile) {
+  constexpr const char kTestMessageStr[] = "Test";
+  constexpr const int expected_splits = 4;
+  constexpr const int bagfile_split_size = 4 * 1024 * 1024;  // 4MB.
+  constexpr const int message_size = 1024 * 1024;  // 1MB
+
+  auto message = get_messages_strings()[0];
+
+  {
+    const auto message_contents_length = static_cast<int>(strlen(kTestMessageStr));
```

Comment:
would `const int` not do the cast for you?

Reply:
`static_cast<int>` performs an explicit cast, while what you suggested performs an implicit cast.